### PR TITLE
[diem framework] Add gas constant setters

### PR DESCRIPTION
**Pull request by tzakian**
This PR adds an additional script function, and function in `DiemVMConfig` to update the gas constants used by the VM for gas metering.

### Testing

An additional test has been added to test that:
* Only the DR account can update the gas constants
* Basic consistency checks are being performed
* That the newly updated gas constants are being used by the VM

The new function has been specified. Sadly I couldn't use the `DiemConfig::SetConfig` schema since the prover didn't like the usage of `old` within that (but maybe I missed something).

### CODE REVIEW

File: language/diem-framework/modules/DiemVMConfig.move [hunk 1]
```diff
@@ -134,6 +138,79 @@ module DiemVMConfig {
...
+        gas_constants.max_price_per_gas_unit            = max_price_per_gas_unit;
+        gas_constants.max_transaction_size_in_bytes     = max_transaction_size_in_bytes;
+        gas_constants.gas_unit_scaling_factor           = gas_unit_scaling_factor;
+        gas_constants.default_account_size              = default_account_size;
+
+        DiemConfig::set(dr_account, config);
+    }
+    spec fun set_gas_constants {
+        include DiemTimestamp::AbortsIfNotOperating;
+        /// No one can update DiemVMConfig except for the Diem Root account [[H11]][PERMISSION]
```
Comment (tzakian) [inline language/diem-framework/modules/DiemVMConfig.move:185]: This follows the permission module in DIP-2, cc @junkil-park just to double-check/FYI as you're auditing this at the moment.

**Comment by bob-wilson** [comment]
+1 Thanks, Tim!

**Comment by tnowacki** [comment]
You will have to rebase on #7949 if it ever lands...

File: language/diem-framework/modules/DiemVMConfig.move (hunk 1 above)
Comment (junkil-park) [reply language/diem-framework/modules/DiemVMConfig.move:185]: LGTM. Nit: could you add a period "." at the end of the sentence?

**Review by runtian-zhou — APPROVED** [review]
LGTM!

**Pull request closed without merging**